### PR TITLE
Adding functionality for getting transactions for league  

### PR DESCRIPTION
**Pull request by DMcP89**

### CODE REVIEW

File: yahoo_fantasy_api/league.py [hunk 1]
```diff
@@ -733,6 +733,35 @@ def draft_results(self):
             dres.append(p)
         return(dres)
 
+    def transactions(self, tran_types, count):
+        '''
+        Fetch transactions of a given type for the league.
+
+        :param tran_types: The comman seperated types of transactions retrieve.  Valid values
+        are: add,drop,commish,trade
```
Comment (spilchen) [inline yahoo_fantasy_api/league.py:741]: This need to indented as it messes up the docs.  sphinx doesn't know that this line is a continuation of the `tran_types` parameter.  Similar comment for `count` parameter.

File: yahoo_fantasy_api/tests/test_league.py [hunk 2]
```diff
@@ -215,3 +215,11 @@ def test_player_details_cache(mock_nhl_league):
...
     exp_keys = ['Phil', 3983, 5085, 5387]
     assert(list(mock_nhl_league.player_details_cache.keys()) == exp_keys)
+
+
+def test_transactions(mock_mlb_league):
+    transactions = mock_mlb_league.transactions("trade", "1")
+    for transaction in transactions:
+        assert(transaction)
+        assert(transaction['type'] == 'trade')
+        assert(transactions['players'])
```
Comment (spilchen) [inline yahoo_fantasy_api/tests/test_league.py:225]: There was an error in the travis build here.  Should this be `transaction` (i.e. drop the **s**)?

**Review by spilchen — CHANGES_REQUESTED** [review]
I like the new API.  Be sure to update the version to 2.4.0 so I can create a new python package.

**Review by spilchen — APPROVED** [review]

**Comment by curropar** [comment]
Hi, does this new API allow to retrieve `pending_trade` with status `accepted`?  The goal would be to report of a new trade accepted by both parties, that the Commissioner has yet to approve, or the Managers have to vote (depending on League settings).

**Comment by DMcP89** [comment]
@curropar 
I'm actually working on a similar use case and was able to figure it out. In order to get transactions marked as pending trades you'll want to use the Team object instead of League.

If you look at Yahoo's documentation - https://developer.yahoo.com/fantasysports/guide/#transactions-collection

Type with team_key | waiver,pending_trade | You can only use these options when also providing the team_key, ie /transactions;team_key=257.l.193.t.1;type=waiver
-- | -- | --

So in order to get pending_trade transactions with status accepted you'll want to do something like this:

```
team = league.to_team(key)
accepted_trades = list(filter(lambda d: d['status'] == 'accepted', team.proposed_trades()))

```

Pending_trades that have been accepted but have not been approved or voted on are available to all teams in the league so you can use which ever team key you want. Take this request:

`https://fantasysports.yahooapis.com/fantasy/v2/league/403.l.91853/transactions;team_key=403.l.91853.t.2;type=pending_trade`

As you can see the team key we're providing here is 403.l.91853.t.2, now notice the team keys in the response:
```

{
  "fantasy_content": {
    "xml:lang": "en-US",
    "yahoo:uri": "\/fantasy\/v2\/league\/403.l.91853\/transactions;team_key=403.l.91853.t.2;type=pending_trade",
    "league": [
      {
        "league_key": "403.l.91853",
        "league_id": "91853",
        "name": "Test League",
        "url": "https:\/\/hockey.fantasysports.yahoo.com\/hockey\/91853",
        "logo_url": false,
        "draft_status": "postdraft",
        "num_teams": 4,
        "edit_key": "2021-01-11",
        "weekly_deadline": "intraday",
        "league_update_timestamp": null,
        "scoring_type": "headpoint",
        "league_type": "private",
        "renew": "",
        "renewed": "",
        "iris_group_chat_id": "",
        "allow_add_to_dl_extra_pos": 1,
        "is_pro_league": "0",
        "is_cash_league": "0",
        "current_week": "1",
        "start_week": "1",
        "start_date": "2021-01-13",
        "end_week": "16",
        "end_date": "2021-05-08",
        "game_code": "nhl",
        "season": "2020"
      },
      {
        "transactions": {
          "0": {
            "transaction": [
              {
                "transaction_key": "403.l.91853.pt.1",
                "type": "pending_trade",
                "status": "accepted",
                "trader_team_key": "403.l.91853.t.4",
                "trader_team_name": "Dave's Dandy Team2",
                "tradee_team_key": "403.l.91853.t.1",
                "tradee_team_name": "Dave's Dandy Team",
                "trade_proposed_time": "1610373048",
                "tradee_unused_roster_spots": 0,
                "current_login_can_respond": 0,
                "trade_accepted_time": "1610414757",
                "trade_reject_time_days": "2",
                "trade_process_date": "2021-01-14",
                "trade_roster_reflect_key": "2021-01-14"
              },
.......
```
The team keys for this trade are 403.l.91853.t.4 & 403.l.91853.t.1, so as long as you have a valid team key you'll be able to get the accepted trades.

Let me know if this answers you question or if there's something you're not clear on.

**Comment by curropar** [comment]
Thanks! That was exactly what I was looking for! I'm new to Python and didn't know about the `list(filter())`, so I was looking for, well, not appropriate solutions. And that strange documentation from Yahoo certainly does not help. Thanks again!

**Comment by curropar** [comment]
Hi again, 
Although I'm implementing this in my bot, I may suggest to implement it in this project, so you may call it directly with `league.pending_trades()`, or maybe another name like `league.accepted_trades()` so it's not confused with Yahoo's API nomenclature. 
I would gladly do it myself, but this honestly goes beyond my coding skills. On the other hand, I might help for testing or whatever is needed. 
Cheers.

**Pull request merged**